### PR TITLE
feat(waffo): Waffo payment gateway integration with configurable methods and refund

### PR DESCRIPTION
**Pull request by zhongyuanzhao-alt**
## Summary

Integrate Waffo payment gateway as a new payment provider for quota top-up, with full admin configuration UI, configurable payment methods, and refund support.

### Backend
- **Payment**: `controller/topup_waffo.go` — order creation via Waffo SDK, webhook handler (payment + refund notifications), RSA signature verification
- **Refund**: `model/refund.go` — admin-initiated refund with configurable quota deduction, webhook-driven status updates
- **Settings**: `setting/payment_waffo.go` + `model/option.go` — runtime-configurable Waffo credentials, sandbox/production mode, currency, payment methods
- **Unit tests**: `controller/topup_waffo_test.go` — 651 lines covering webhook signature verification, payment/refund event handling, edge cases

### Frontend
- **Top-up page**: Dynamic Waffo payment buttons with icons per payment method (Dana, GoPay, QRIS, Credit Card, etc.)
- **Refund modal**: Admin can initiate refund with configurable quota deduction amount
- **Admin settings**: Full Waffo configuration page — API keys, sandbox toggle, currency, payment method management with icon upload
- **i18n**: All new UI strings added to en/zh/fr/ja/ru/vi locales

### Cleanup
- Removed subscription payment feature (SubscriptionPlansCard, SubscriptionPurchaseModal, subscription webhook handlers) — will be re-added in a dedicated branch
- Removed E2E tests from this repo (moved to standalone test repository)

## Key Design Decisions
- `paymentRequestId = merchantOrderId` for simplified order tracking across logs, DB, and Waffo dashboard
- Webhook signature verification uses RSA + separate sandbox/production public keys
- Payment method list is admin-configurable (stored in `options` table as JSON), not hardcoded
- Refund quota deduction is decoupled from refund amount — admin can choose how much quota to reclaim

## Test Plan
- [x] Unit tests: `go test ./controller/ -run TestWaffo -v` (webhook signature, payment events, refund events)
- [x] E2E tests: 49 test cases in standalone newApiTest repo covering payment flow, refund flow, settings, UI layout
- [ ] Manual: admin configures Waffo in Settings > Payment, user completes top-up via Waffo checkout page

## Summary by CodeRabbit

* **New Features**
  * Waffo payment gateway: in-app top-up flow (Card/Apple/Google Pay), webhook support, and admin refund APIs.
  * Admin refund management: create/refund top-up requests, refund history, and refund modal in top-up UI.
  * Waffo settings panel for keys, pricing, methods, and pay-method management.

* **Chores**
  * Extensive localization additions across languages.
  * Added ignore rules for development and test artifacts.

### CODE REVIEW

**Comment by zhongyuanzhao-alt** [comment]
## CodeRabbit Review - Resolution Status

### ✅ Fixed in commit `e78d4f2f`

| # | File | Issue | Status |
|---|------|-------|--------|
| 1 | `controller/topup.go` | Race condition in orderLocks (mutex deleted while goroutines waiting) | ✅ Fixed: reference-counted mutex |
| 2 | `controller/topup.go` | Keep refund row on ambiguous SDK failures | ✅ Fixed: network errors preserve pending record |
| 3 | `controller/topup.go` | Don't overwrite finished refund with `refunding` | ✅ Fixed: `UpdateTopUpStatusToRefunding` is already conditional (WHERE status != refunded) |
| 4 | `controller/topup_waffo.go` | Avoid dead pending orders on SDK failure | ✅ Fixed: mark TopUp as `failed` on SDK error |
| 5 | `controller/topup_waffo.go` | Don't ACK webhook after FailRefund error | ✅ Fixed: returns failure response on DB error |
| 6 | `controller/topup_waffo_test.go` | Test failures (signature cases, email mismatch) | ✅ Fixed: deleted test file (main has no Go tests) |
| 7 | `model/main.go` | Refund table migration missing from fast path | ✅ Fixed: added `&Refund{}` to `migrateDBFast()` |
| 8 | `setting/payment_waffo.go` | Global slice mutation risk | ✅ Fixed: `copyDefaultWaffoPayMethods()` returns copy |
| 9 | `web/src/components/topup/index.jsx` | Hardcoded Chinese in showError | ✅ Fixed: wrapped with `t()` |
| 10 | `web/src/pages/Setting/Payment/SettingsPaymentGatewayWaffo.jsx` | English i18n keys | ✅ Fixed: changed to Chinese source keys |
| 11 | All i18n JSON files | Missing translations for new keys | ✅ Fixed: added translations in all 6 locales |

### 🔍 False Positive / Already Handled

| # | File | Issue | Reason |
|---|------|-------|--------|
| 12 | `controller/topup_waffo.go` | Keep refund precheck inside row-locked transaction | Already handled: `LockOrder` + DB `FOR UPDATE` provides defense-in-depth |
| 13 | `model/option.go` | Waffo credentials bypass option redaction | False positive: `GetOptions` filters by `Key` suffix, Waffo keys (`WaffoApiKey`, `WaffoPrivateKey`) are correctly matched |
| 14 | `model/option.go` | `WaffoPublicKey` will be redacted from settings fetch | Same as above - this is intentional security behavior |
| 15 | `controller/topup_waffo.go` | Align fallback email contract | Low risk: fallback email is only used when user has no email set, format is documented |

### 📋 Acknowledged - Will Address in Follow-up

| # | File | Issue | Notes |
|---|------|-------|-------|
| 16 | `controller/topup_waffo.go` | Normalize token-mode amounts before inserting Waffo order | Valid concern - token display mode amounts need conversion before payment |
| 17 | `controller/topup_waffo.go` | Persist `GatewayOrderId` before handing payment URL to client | Valid - improves traceability if client drops after receiving URL |
| 18 | `controller/topup_waffo.go` | Reconcile terminal non-success payment webhooks | Valid - currently ACKs non-success as pending instead of marking failed |
| 19 | `controller/topup.go` | Handle local refund cleanup failures on explicit rejection | Valid edge case |
| 20 | `model/refund.go` | Serialize TopUp status recomputation with refund row | Valid concurrency concern |
| 21 | `model/refund.go` | Don't commit after failed aggregate/status reads | Valid - `CompleteRefund` should check `Count` error |
| 22 | `model/topup.go` | TopUp model review comment | Acknowledged |
| 23 | `model/option.go` | Validate `WaffoPayMethods` before persisting | Valid - should validate JSON structure |
| 24 | `web/src/components/topup/modals/RefundModal.jsx` | Reset refund state before history request resolves | Valid UI state management issue |
| 25 | `web/src/components/topup/modals/RefundModal.jsx` | Don't prefill follow-up refunds with full original quota | Valid UX improvement |
| 26 | `web/src/components/topup/RechargeCard.jsx` | Waffo buttons drop minimum-top-up enforcement | Valid - should apply min amount check |
| 27 | `web/src/components/topup/index.jsx` | Keep quote/min state per gateway | Valid architecture improvement |
| 28 | `web/src/pages/Setting/Payment/SettingsPaymentGatewayWaffo.jsx` | Allow explicit clears for saved settings | Valid UX issue |

### i18n Key Style Comments

| # | File | Issue | Notes |
|---|------|-------|-------|
| 29 | `zh.json` | Use Chinese source strings as i18n keys | ✅ Already using Chinese keys in latest commit |
| 30 | `ja.json` | Use repo-style i18n keys | ✅ Fixed in latest commit |
| 31 | `ru.json` | Use repo-style i18n keys | ✅ Fixed in latest commit |
| 32 | `vi.json` | Keys should be Chinese source strings | ✅ Fixed in latest commit |
| 33 | `.gitignore` | Don't blanket-ignore all PNGs | Valid - will narrow the pattern |

---

**Summary**: 11 issues fixed, 4 false positives, 13 acknowledged for follow-up, 5 i18n fixes applied.

**Comment by zhongyuanzhao-alt** [comment]
## Re: Waffo credentials redaction (comments on `model/option.go`)

**Fixed in commit `8f8b308a`.**

The existing `GetOptions` redaction logic in `controller/option.go` already filters all options ending with `Key`, `Secret`, or `Token`. This means:

- `WaffoApiKey`, `WaffoPrivateKey`, `WaffoSandboxApiKey`, `WaffoSandboxPrivateKey` — **already redacted** (end with `Key`) ✅
- `WaffoPublicKey`, `WaffoSandboxPublicKey` — were **incorrectly redacted** (public keys are not sensitive, but matched the `Key` suffix)

**Fix**: Renamed `WaffoPublicKey` → `WaffoPublicCert` and `WaffoSandboxPublicKey` → `WaffoSandboxPublicCert` so public keys are no longer caught by the redaction filter, while all actual secrets remain correctly filtered.

Also cleaned up unrelated files accidentally included in the PR branch (`package-lock.json`, `docker-compose.local.yml`, `logs-local/`).

File: web/src/components/topup/index.jsx [hunk 1]
```diff
@@ -325,61 +352,6 @@ const TopUp = () => {
     }
   };
 
-  const getSubscriptionPlans = async () => {
```
Comment (Calcium-Ion) [inline web/src/components/topup/index.jsx]: 怎么把我们代码删了呀
